src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp

```diff
@@ -285,18 +296,174 @@ ZeroInferRequest::ZeroInferRequest(const std::shared_ptr<ZeroInitStructsHolder>&
         // Only one buffer per state variable is required, we'll use the "output" one since this one captures the latest
         // tensor value
         allocate_tensor(stateName, stateDescriptor, TensorType::State, allocator);
+        _tensorsData[stateInputBufferName] = TensorData{_copyAllTensors.at(stateInputBufferName)->data(),
```

Comment:
not really.  allocate_tensor is part of the SyncInferRequest class since tensorData is part only of the Level Zero Infer Request. We don't need such a structure for other backends.